src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,237 @@
+using AWS.Deploy.Orchestration.Utilities;
+using AWS.Deploy.Recipes;
+
+namespace AWS.Deploy.CLI.Commands
+{
+    /// <summary>
+    /// The class supports the functionality to create a new CDK project and save it at a customer
+    /// customer specified directory location.
+    /// </summary>
+    public class GenerateDeploymentProjectCommand
+    {
+        private readonly IToolInteractiveService _toolInteractiveService;
+        private readonly IConsoleUtilities _consoleUtilities;
+        private readonly ICdkProjectHandler _cdkProjectHandler;
+        private readonly ICommandLineWrapper _commandLineWrapper;
+        private readonly OrchestratorSession _session;
+
+        public GenerateDeploymentProjectCommand(
+            IToolInteractiveService toolInteractiveService,
+            IConsoleUtilities consoleUtilities,
+            ICdkProjectHandler cdkProjectHandler,
+            ICommandLineWrapper commandLineWrapper,
+            OrchestratorSession session)
+        {
+            _toolInteractiveService = toolInteractiveService;
+            _consoleUtilities = consoleUtilities;
+            _cdkProjectHandler = cdkProjectHandler;
+            _commandLineWrapper = commandLineWrapper;
+            _session = session;
+        }
+
+        /// <summary>
+        /// This method takes a user specified directory path and generates the CDK deployment project at this location.
+        /// If the provided directory path is an empty string, then a default directory is created to save the CDK deployment project.
+        /// </summary>
+        /// <param name="saveCdkDirectoryPath">An absolute or a relative path provided by the user.</param>
+        /// <returns></returns>
+        public async Task ExecuteAsync(string saveCdkDirectoryPath)
+        {
+            var orchestrator = GetOrchestrator();
+            var recommendations = await GenerateDeploymentRecommendations(orchestrator);
+            var selectedRecommendation = _consoleUtilities.AskToChooseRecommendation(recommendations);
+
+            if (string.IsNullOrEmpty(saveCdkDirectoryPath))
+                saveCdkDirectoryPath = GenerateDefaultSaveDirectoryPath(_session.ProjectDefinition.ProjectPath);
+
+            var newDirectoryCreated = CreateSaveCdkDirectory(saveCdkDirectoryPath);
+
+            var errorMessage = ValidateSaveCdkDirectory(saveCdkDirectoryPath);
+            if (!string.IsNullOrEmpty(errorMessage))
+            {
+                if (newDirectoryCreated)
+                    Directory.Delete(saveCdkDirectoryPath);
+                throw new InvalidSaveDirectoryForCdkProject(errorMessage);
+            }
+
+            var directoryUnderSourceControl = await IsDirectoryUnderSourceControl(saveCdkDirectoryPath);
+            if (!directoryUnderSourceControl)
+            {
+                _toolInteractiveService.WriteLine();
+                var yesNoResult = _consoleUtilities.AskYesNoQuestion("The target directory for saving the CDK " +
```

Comment:
If we are going to have this warning then we should explain why it is necessary. For example we want to explain that it is important that project is preserved if it is used for deployment to support redeployments. We should double check with Irene this experience.

Reply:
Changed the warning to reflect the above.